### PR TITLE
Problem: empty topics in rpc response is not tested

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #841

Solution:
- merge empty topics fix
- add integration tests to verify

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by calvinaco — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -161,4 +161,4 @@ replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.3
 
 // https://github.com/crypto-org-chain/ethermint/tree/v0.9.x-cronos
 // FIXME make a tag or point to upstream when ready
-replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.6.1-0.20211207040748-254df3803d62
+replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.6.1-0.20211214070921-bfea18622b20
```
Comment (yihuang) [inline go.mod:164]: better wait for upstream pr https://github.com/tharsis/ethermint/pull/840 merged first.

Comment (yihuang) [reply go.mod]: merged now.

**Comment by thomas-nguy** [comment]
to make this change more clear, can we separate the ethermint version increase in a different PR?

**Comment by yihuang** [comment]
integration test failure is due to this issue: https://github.com/tharsis/ethermint/pull/851

**Comment by yihuang** [comment]
> to make this change more clear, can we separate the ethermint version increase in a different PR?

sounds good.

**Pull request merged**